cuqi/solver/_solver.py

```diff
@@ -147,18 +139,20 @@ def solve(self):
         info = {"success": solution['success'],
                 "message": solution['message'],
                 "func": solution['fun'],
-                "grad": solution['jac'],
                 "nit": solution['nit'], 
                 "nfev": solution['nfev']}
+        # if gradfunc is callable, record the gradient in the info dict
+        if callable(self.gradfunc):
+            info['grad'] = solution
```

Comment:
- Thank you Amal for pointing the typo!
- If we don't provide the jac routine, `scipy.optimize.minimize` will actually not record `jac` in the dictionary, in contrast to `fmin_l_bfgs_b`: this is indeed strange behaviour and that's why the code is `ScipyMinimizer` is slightly different from `ScipyLBFGSB`. See the following MWE, there is no `jac` in the keys of the returned solution object.
```python
from scipy.optimize import minimize, rosen, rosen_der
x0 = [1.3, 0.7, 0.8, 1.9, 1.2]
res = minimize(rosen, x0, method='Nelder-Mead', tol=1e-6)
print(res.keys())
```
Output
```bash
dict_keys(['fun', 'nit', 'nfev', 'status', 'success', 'message', 'x', 'final_simplex'])
```

Reply:
Hi Charlie, thank you for the clarification and the nice MWE. 
I believe not having 'jac' recorded is because of the optimization method choice and not because of not supplying the gradient. for example 
```
from scipy.optimize import minimize, rosen
x0 = [1.3, 0.7, 0.8, 1.9, 1.2]
res = minimize(rosen, x0, tol=1e-6)
print(res.keys())
```
gives
```
dict_keys(['fun', 'jac', 'hess_inv', 'nfev', 'njev', 'status', 'success', 'message', 'x', 'nit'])
```

Reply:
Thank you, Amal. It should be based on the method instead of if the gradient is supplied. In the end, I think it is probably easier just to check if the key `jac` exists in the returned dictionary: if it `jac` exits, record it in info; otherwise, do nothing.